bundles/org.openhab.binding.sedif/src/main/java/org/openhab/binding/sedif/internal/handler/ThingSedifHandler.java

```diff
@@ -0,0 +1,698 @@
+                        // We just return null, EpiringDayCache logic will retry the operation later.
+                        logger.error("Failed to get contract details", ex);
+                        return null;
+                    } catch (InvalidSessionException ex) {
+                        // In case of session error, we force the bridge to reconnect after a delay
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+
+                        Bridge lcBridge = getBridge();
+                        if (lcBridge != null && lcBridge.getHandler() instanceof BridgeSedifWebHandler bridgeSedif) {
+                            bridgeSedif.scheduleReconnect();
+                        }
+                        return null;
+                    } catch (SedifException ex) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                ex.getMessage());
+                        return null;
+                    }
+                });
+
+        this.consumption = new ExpiringDayCache<MeterReading>("dailyConsumption", REFRESH_HOUR_OF_DAY,
+                REFRESH_MINUTE_OF_DAY, () -> {
+                    LocalDate today = LocalDate.now();
+
+                    try {
+                        MeterReading meterReading = updateConsumptionData(today.minusDays(HISTORICAL_LOOKBACK_DAYS),
+                                today, false);
+                        if (meterReading != null) {
+                            MeterReadingHelper.calcAgregat(meterReading);
+                        }
+                        return meterReading;
+                    } catch (CommunicationFailedException ex) {
+                        // We just return null, EpxiringDayCache logic will retry the operation later.
+                        logger.warn("Failed to get consumption data", ex);
+                        return null;
+                    } catch (InvalidSessionException ex) {
+                        // In case of session error, we force the bridge to reconnect after a delay
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        Bridge lcBridge = getBridge();
+                        if (lcBridge != null && lcBridge.getHandler() instanceof BridgeSedifWebHandler bridgeSedif) {
+                            bridgeSedif.scheduleReconnect();
+                        }
+                        return null;
+                    } catch (SedifException ex) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                ex.getMessage());
+                        return null;
+                    }
+                });
+
+        config = getConfigAs(SedifConfiguration.class);
+    }
+
+    @Override
+    public synchronized void initialize() {
+        loadSedifState();
```

Comment:
`config` nneds to be retrieved in `initialize`. So please move the code at line 177 at first line of method `initialize`.
And replace at beginning of class:
```
protected SedifConfiguration config;
```
by
```
protected SedifConfiguration config = new SedifConfiguration();
```
To be checked if the variable really needs to be protected or if it can be private.